src/torchcodec/_core/CMakeLists.txt

```diff
@@ -158,14 +195,17 @@ function(make_torchcodec_libraries
         PRIVATE
         PYBIND_OPS_MODULE_NAME=core_pybind_ops
     )
-    # If we don't make sure this flag is set, we run into segfauls at import
-    # time on Mac. See:
-    #    https://github.com/pybind/pybind11/issues/3907#issuecomment-1170412764
-    target_link_options(
-        ${pybind_ops_library_name}
-        PUBLIC
-        "LINKER:-undefined,dynamic_lookup"
-    )
+
+    if(APPLE)
+        # If we don't make sure this flag is set, we run into segfauls at import
+        # time on Mac. See:
+        # https://github.com/pybind/pybind11/issues/3907#issuecomment-1170412764
+        target_link_options(
+            ${pybind_ops_library_name}
+            PUBLIC
+            "LINKER:-undefined,dynamic_lookup"
+        )
+    endif()
```

Comment:
Sort of a drive-by, but this shouldn't be needed for linux either, and those linker options won't work on windows anyway.

Reply:
Agreed, and I think you're applying a good principle: platform-specific tweaks should have a platform-specific guard.